### PR TITLE
Habit Tracker project

### PR DESCRIPTION
**Pull request by JsPeanut**

### CODE REVIEW

**Comment by TheCSharpAcademy** [comment]
@JsPeanut HI there! Just checking if you have submitted the project in the website? I can't see it in our database.

**Comment by JsPeanut** [comment]
No I haven't. My bad. Will submit it as soon as I get home.

**Comment by JsPeanut** [comment]
I just submitted! Sorry for the delay.

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 1]
```diff
@@ -0,0 +1,405 @@
+using Microsoft.Data.Sqlite;
+using System.Globalization;
+using System.Text.RegularExpressions;
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:3]: Please remove unused using

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+                tableCmd.CommandText = "CREATE TABLE IF NOT EXISTS habits(Id INTEGER PRIMARY KEY AUTOINCREMENT, HabitName TEXT, Date TEXT, MeasurementUnit TEXT, Quantity INTEGER)";
+
+                tableCmd.ExecuteNonQuery();
+
+                connection.Close();
+            }
+
+            GetUserInput();
+        }
+
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:27]: Please remove unnecessary empty line 27

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+                }
+
+                connection.Close();
+
+                foreach (var habit in Habits)
+                {
+                    Console.WriteLine($"{habit.Id} - {habit.HabitName} - {habit.Date.ToString("dd-MMM-yyyy")} - {habit.Quantity} {habit.MeasurementUnit}");
+
+                }
+
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:212]: Please remove unnecessary empty lines 212, 213

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+                var tableCmd = connection.CreateCommand();
+
+                tableCmd.CommandText = $"UPDATE habits SET HabitName = '{habit}', Date = '{date}', MeasurementUnit = '{measure}', Quantity = {quantity} WHERE Id = {recordId}";
+
+                tableCmd.ExecuteNonQuery();
+
+                connection.Close();
+
+                GetUserInput();
+
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:285]: Please remove unnecessary empty line 285

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+                    else if (leastConsistentHabits.Count() > 1)
+                    {
+                        string cHabitString = string.Join(", ", leastConsistentHabits);
+                        Console.WriteLine($"The habits you were the least consistent with were '{cHabitString}' with a {leastConsistentHabits_}-day streak.");
+                    }
+                    connection.Close();
+
+                    GetUserInput();
+                }
+
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:386]: Please remove unnecessary empty line 386

File: HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+            }
+        }
+
+        class Habit
+        {
+            public int Id { get; set; }
```
Comment (TheCSharpAcademy) [inline HabitTracker.JsPeanut/HabitTracker.JsPeanut/Program.cs:392]: For consistency, either have one empty line between properties, or none.

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]
Hi! The project looks great, well done! 

All the comments are in regards to code tidiness. Usually I don't mind these in the initial projects, but since is your third, I'll be enforcing our code conventions, which can be found here: http://thecsharpacademy.com/article/58

**Comment by TheCSharpAcademy** [comment]
@JsPeanut let me know if you need help! 😁

**Comment by JsPeanut** [comment]
sorry man! i was too busy trying to finish the coding tracker project that i forgot this one. done

**Pull request merged**